### PR TITLE
[Hub Generated] Review request for Microsoft.Aadiam to add version stable/2018-01-01

### PR DESCRIPTION
**Pull request by jssmomo**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/jssmomo/azure-rest-api-specs/dev-azureactivedirectory-Microsoft.Aadiam-2018-01-01...master/).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/azureactivedirectory/resource-manager/readme.md [hunk 1]
```diff
@@ -24,16 +24,25 @@ For other options on installation see [Installing AutoRest](https://aka.ms/autor
...
 ```
 
+
+### Tag: package-2018-01
+
+These settings apply only when `--tag=package-2018-01` is specified on the command line.
+
+```yaml $(tag) == 'package-2018-01'
+input-file:
+  - Microsoft.Aadiam/stable/2018-01-01/azureactivedirectory.json
```
Comment (ArcturusZhang) [inline specification/azureactivedirectory/resource-manager/readme.md]: You should put the swagger files `metrics_API.json` and `metricDefinitions_API.json` in this section

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]
Please fix the errors that CI reports

**Comment by jssmomo** [comment]
> Please fix the errors that CI reports

Hi ArcturusZhang, can you help me fix the issues? Most of the issues are similar but I could not figure out what could be the root cause. The error is:

scripts/multiapi.ts(95,40): error TS2345: Argument of type 'Set<string>' is not assignable to parameter of type 'Iterable<any>'.

**Comment by ArcturusZhang** [comment]
> > Please fix the errors that CI reports
> 
> Hi ArcturusZhang, can you help me fix the issues? Most of the issues are similar but I could not figure out what could be the root cause. The error is:
> 
> scripts/multiapi.ts(95,40): error TS2345: Argument of type 'Set' is not assignable to parameter of type 'Iterable'.

That is caused by a bug introduced by typescript update, which is resolved now.
As for the failure on SDK python, I will try to consult the SDK owner of python

**Comment by ArcturusZhang** [comment]
Hi @zikalino could you please help us check the failure on SDK python check?

**Comment by jssmomo** [comment]
> Hi @zikalino could you please help us check the failure on SDK python check?

Hi @ArcturusZhang I fixed the validation. Can you let me know what is the next steps?

**Comment by ArcturusZhang** [comment]
> > Hi @zikalino could you please help us check the failure on SDK python check?
> 
> Hi @ArcturusZhang I fixed the validation. Can you let me know what is the next steps?

Now we just need waiting for feedback from the ARM team, I will write an email to inform them of this pr

**Comment by jssmomo** [comment]
Some more context: the added Metrics & MetricDefinitions APIs are the Tenant-Level equivalent of the existing Monitoring APIs. The schema and parameters are highly similar to the existing Monitoring APIs

**Comment by KrisBash** [comment]
@jssmomo - do you want to move this to the private github repo? My understanding from discussion with the Insights team is that this is a prototype that will not ship publicly.

**Comment by jssmomo** [comment]
> @jssmomo - do you want to move this to the private github repo? My understanding from discussion with the Insights team is that this is a prototype that will not ship publicly.

I think it make sense. One question I have before creating PR in private repo is that for our RP, we didn't work with private repo when we creating the specs. So should I also include the older APIs in the PR?

**Comment by ArcturusZhang** [comment]
Hi @jssmomo will there be more update to this pr? If not, I am going to close this pr for now.

**Comment by jssmomo** [comment]
> Hi @jssmomo will there be more update to this pr? If not, I am going to close this pr for now.

Hi @ArcturusZhang, please let me confirm with Kris. I'll let you know the update early next week.

**Comment by jssmomo** [comment]
Closing the PR as I created one in Private repo:
https://github.com/Azure/azure-rest-api-specs-pr/pull/935

**Pull request closed without merging**